build.rs

```diff
@@ -269,7 +269,8 @@ fn main() {
     // Avoid assuming the working directory is the same is the $CARGO_MANIFEST_DIR so that toolchains
     // which may assume other working directories can still build this code.
     let c_root_dir = PathBuf::from(
-        env::var_os("CARGO_MANIFEST_DIR").expect("CARGO_MANIFEST_DIR should always be set"),
+        // Don't emit `cargo:rerun-if-env-changed` for $CARGO_MANIFEST_DIR.
+        std::env::var_os("CARGO_MANIFEST_DIR").expect("CARGO_MANIFEST_DIR should always be set"),
```

Comment:
I suggest that instead of doing this, we change the implementation of the local `env::var_os`/`env::var` to take an `&'static EnvVar` instead of a `&str` for the environment variable name:

```rust
struct EnvVar {
    name: &'static str,
    ty: EnvVarType
}
enum EnvVarType {
    RerunIfChanged,
    CargoSetEnvVar,
}
const CARGO_MANIFEST_DIR: EnvVarName = EnvVarName {
    name: "CARGO_MANIFEST_DIR",
    ty: CargoSetEnvVar,
};
...
```

Otherwise, we'd have to repeat the pattern you have above for every Cargo-set environment variable, apparently.

Reply:
Thanks for the review! I've updated the PR, I hope I've understood your comment correctly.